Modules/Filtering/ImageFilterBase/include/itkTernaryGeneratorImageFilter.h

```diff
@@ -0,0 +1,277 @@
+
+  /** Connect one of the operands for pixel-wise operation. */
+  void
+  SetInput1(const TInputImage1 * image1);
+  virtual void
+  SetInput1(const DecoratedInput1ImagePixelType * input1);
+  virtual void
+  SetInput1(const Input1ImagePixelType & input1);
+
+  /** Set the first operand as a constant. */
+  virtual void
+  SetConstant1(const Input1ImagePixelType & input1);
+
+  /** Get the constant value of the first operand. An exception is sent if
+   * the first operand is not a constant. */
+  virtual const Input1ImagePixelType &
+  GetConstant1() const;
+
+  /** Connect one of the operands for pixel-wise operation. */
+  void
+  SetInput2(const TInputImage2 * image2);
+  virtual void
+  SetInput2(const DecoratedInput2ImagePixelType * input2);
+  virtual void
+  SetInput2(const Input2ImagePixelType & input2);
+
+  /** Set the second operand as a constant. */
+  virtual void
+  SetConstant2(const Input2ImagePixelType & input2);
+
+  /** Get the constant value of the second operand. An exception is sent if
+   * the second operand is not a constant. */
+  virtual const Input2ImagePixelType &
+  GetConstant2() const;
+
+  /** Connect one of the operands for pixel-wise operation. */
+  void
+  SetInput3(const TInputImage3 * image3);
+  virtual void
+  SetInput3(const DecoratedInput3ImagePixelType * input3);
+  virtual void
+  SetInput3(const Input3ImagePixelType & input3);
+
+  /** Set the second operand as a constant. */
+  virtual void
+  SetConstant3(const Input3ImagePixelType & input3);
+
+  /** Get the constant value of the second operand. An exception is sent if
+   * the second operand is not a constant. */
+  virtual const Input3ImagePixelType &
+  GetConstant3() const;
+
+
+#if !defined(ITK_WRAPPING_PARSER)
+  /** Set the pixel functor
+   *
+   * The functor defines an operation done per pixel.
+   */
+  void
+  SetFunctor(const std::function<ConstRefFunctionType> & f)
+  {
+    // the closure create a copy of f
+    m_DynamicThreadedGenerateDataFunction = [this, f](const OutputImageRegionType & outputRegionForThread) {
+      return this->DynamicThreadedGenerateDataWithFunctor(f, outputRegionForThread);
+    };
+
+    this->Modified();
+  }
+  void
+  SetFunctor(const std::function<ValueFunctionType> & f)
+  {
+    // the capture create a copy of f
+    m_DynamicThreadedGenerateDataFunction = [this, f](const OutputImageRegionType & outputRegionForThread) {
+      return this->DynamicThreadedGenerateDataWithFunctor(f, outputRegionForThread);
+    };
+
+    this->Modified();
+  }
+
+  /** Set the pixel functor by a C function pointer
+   *
+   * The functor defines an operation done per pixel.
+   */
+  void
+  SetFunctor(ConstRefFunctionType * funcPointer)
+  {
+    m_DynamicThreadedGenerateDataFunction = [this, funcPointer](const OutputImageRegionType & outputRegionForThread) {
+      return this->DynamicThreadedGenerateDataWithFunctor(funcPointer, outputRegionForThread);
+    };
+
+    this->Modified();
+  }
+
```

Comment:
No empty lines here so that doxygen entry for above method also applies for the one below.